### PR TITLE
Add ability to wait for fonts to load with `GoogleFonts.pendingFonts`

### PR DESCRIPTION
**Pull request by johnsonmh**
## Description

`GoogleFonts.pendingFonts` returns a `Future` which resolves when requested fonts have finished loading and are ready to be rendered on screen.
 
This can be used in combination with a `FutureBuilder` to avoid visual font swapping

This PR includes these changes:
* updated example to Material 3
* showcase using `FutureBuilder` in example
* added information to README
* improved documentation

## Related Issues
 - Fixes https://github.com/material-foundation/flutter-packages/issues/151

## Tests
 - Added a test

Fixes https://github.com/material-foundation/google-fonts-flutter/issues/151

## Future work
* Consider adding more fine grained control over this feature by providing `preloadX` functions for each text style and text theme. Currently it is not possible to tell _which_ font loads _when_.

### CODE REVIEW

File: lib/src/google_fonts_base.dart [hunk 1]
```diff
@@ -99,7 +107,9 @@ TextStyle googleFontsTextStyle({
     file: fonts[matchedVariant]!,
   );
 
-  loadFontIfNecessary(descriptor);
+  final loadingFuture = loadFontIfNecessary(descriptor);
```
Comment (guidezpl) [inline lib/src/google_fonts_base.dart:110]: test cases would be nice to validate what this behaves like when loading the font isn't necessary, for instance

**Review by guidezpl — APPROVED** [review]
Otherwise LGTM as an initial functionality. Brain is a bit foggy, but can't think of a way to avoid `pendingFontLoadsInternal`

**Comment by clocksmith** [comment]
instead of return a `Future` of `List` of fonts, can this return a `List` of `Future` fonts? Also, consider renaming to `loadedFonts, since they aren't necessarily pending when this is called

**Comment by devj3ns** [comment]
This is great and very useful @johnsonmh!

I really look forward to this getting merged and published to pub.dev 😀

**Comment by devj3ns** [comment]
Hey @johnsonmh and @guidezpl, I would really appreciate if someone had the time to finish this PR in the near future. 

It's an awesome feature, and you are doing amazing work 💙😄

**Review by piotrsed — APPROVED** [review]

**Comment by therealjbox** [comment]
Get me out of here!!

**Comment by clocksmith** [comment]
@johnsonmh is this still valid?

**Review by hoshigakikisame — APPROVED** [review]

File: lib/src/google_fonts_base.dart (hunk 1 above)
Comment (esouthren) [reply lib/src/google_fonts_base.dart:109]: I think this has been covered by `load_font_if_necessary_with_local_fonts_test.dart`.

**Review by MuhamedFaraag — APPROVED** [review]

**Comment by guidezpl** [comment]
@esouthren hey, hope all is well! Can you please sign the CLA?

**Review by guidezpl — APPROVED** [review]
I've taken over the PR and made improvements, approving to unblock landing.

**Review by JoseAlba — APPROVED** [review]
LGTM

File: packages/google_fonts/test/google_fonts_text_style_test.dart [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    when(mockHttpClient.gets(any)).thenAnswer((_) async {
+      return http.Response(_fakeResponse, 200);
+    });
+  });
+
+  tearDown(() {
+    clearCache();
+  });
+
+  test('pendingFonts works as expected', () async {
```
Comment (clocksmith) [inline packages/google_fonts/test/google_fonts_text_style_test.dart:64]: minor nit: replace ` test('pendingFonts works as expected' with something like `test('pendingFonts can request 2 fonts and loads them asynchronously'

**Comment by guidezpl** [comment]
I'm going to fix https://github.com/material-foundation/flutter-packages/issues/102 and then release :)

**Pull request merged**